### PR TITLE
Feature: Add jsdoc comments on client

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Adds jsdoc comments on client for generating documentation automatically

Task: [APP-804](https://aragonassociation.atlassian.net/browse/APP-804)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder of the package after the [UPCOMING] title and before the latest version.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: javascript/modules/client/src/client.ts [hunk 1]
```diff
@@ -119,12 +172,34 @@ export class Client extends ClientCore implements IClient {
...
-
+  /**
+   *
+   *
+   * @private
+   * @param {ICreateParams} params
+   * @return {*}  {AsyncGenerator<DaoCreationStepValue>}
+   * @memberof Client
+   */
   private async *_createDao(
```
Comment (brickpop) [inline javascript/modules/client/src/client.ts:205]: We might not need that on private functions, but it's nice nonetheless

File: javascript/modules/client/src/client.ts [hunk 2]
```diff
@@ -119,12 +172,34 @@ export class Client extends ClientCore implements IClient {
...
+     * Estimates the gas fee of creating a DAO
+     *
+     * @param {ICreateParams} params
+     * @return {*}  {Promise<GasFeeEstimation>}
+     * @memberof Client
+     */
     create: (params: ICreateParams) => this._estimateCreation(params),
+
+    /**
+     * Estimates the gas fee of depositing ether or an ERC20 token into the dao
```
Comment (brickpop) [inline javascript/modules/client/src/client.ts]:
```suggestion
     * Estimates the gas fee of depositing ether or an ERC20 token into the DAO
```

File: javascript/modules/client/src/client.ts [hunk 3]
```diff
@@ -111,6 +156,14 @@ export class Client extends ClientCore implements IClient {
   };
 
   encoding = {
+    /**
+      * Computes the parameters to be given when creating a proposal that withdraws ether or an ERC20 token from the dao
```
Comment (brickpop) [inline javascript/modules/client/src/client.ts]:
```suggestion
      * Computes the payload to be given when creating a proposal that withdraws ether or an ERC20 token from the DAO
```

File: javascript/modules/client/src/client.ts [hunk 4]
```diff
@@ -84,24 +84,69 @@ export class Client extends ClientCore implements IClient {
...
+     * @param {string} daoAddressOrEns
+     * @param {string[]} tokenAddresses
+     * @return {*}  {Promise<AssetBalance[]>}
+     * @memberof Client
+     */
     getBalances: (daoAddressOrEns: string, tokenAddresses: string[] = []): Promise<AssetBalance[]> =>
       this._getBalances(daoAddressOrEns, tokenAddresses),
-    /** Retrieves the list of asset transfers to and from the given DAO, by default, from ETH, DAI, USDC and USDT on Mainnet*/
+    /** 
+     * Retrieves the list of asset transfers to and from the given DAO, by default, from ETH, DAI, USDC and USDT on Mainnet
```
Comment (brickpop) [inline javascript/modules/client/src/client.ts]:
```suggestion
     * Retrieves the list of asset transfers to and from the given DAO (by default, from ETH, DAI, USDC and USDT, on Mainnet)
```

File: javascript/modules/client/src/client.ts [hunk 5]
```diff
@@ -84,24 +84,69 @@ export class Client extends ClientCore implements IClient {
...
+     * Created a DAO with the given parameters and plugins
+     *
+     * @param {ICreateParams} params
+     * @return {*}  {AsyncGenerator<DaoCreationStepValue>}
+     * @memberof Client
+     */
     create: (params: ICreateParams) => this._createDao(params),
-    /** Deposits ether or an ERC20 token */
+    /** 
+     * Deposits ether or an ERC20 token
```
Comment (brickpop) [inline javascript/modules/client/src/client.ts]:
```suggestion
     * Deposits ether or an ERC20 token into the DAO
```

File: javascript/modules/client/src/client.ts [hunk 6]
```diff
@@ -84,24 +84,69 @@ export class Client extends ClientCore implements IClient {
 
   /** Contains all the generic high level methods to interact with a DAO */
   methods = {
-    /** Created a DAO with the given parameters and plugins */
+    /**
+     * Created a DAO with the given parameters and plugins
```
Comment (brickpop) [inline javascript/modules/client/src/client.ts]:
```suggestion
     * Creates a DAO with the given settings and plugins
```

File: javascript/modules/client/src/client.ts (hunk 1 above)
Comment (josemarinas) [reply javascript/modules/client/src/client.ts:205]: I'm going to delete it for consistency, if we want to add jsdoc on private functions it should be a different task, and we should add it everywhere

Comment (brickpop) [reply javascript/modules/client/src/client.ts:205]: I don't see it deleted

**Review by brickpop — APPROVED** [review]

**Pull request merged**